Automattic/delphin
# Update eslint hook

This PR improves the pre-commit hook by only running the linter on modified files.
This is useful when we want to commit some of our files and still have work to do on other files (which are failing on lint).
### Testing instructions
- `git checkout update/eslint-hook`
- `npm install`
- Modify a js file to make it fail eslint
- `git add .`
- `git commit`
- Check that the commit failed
- `git reset --hard HEAD`
- Modify an scss file to make it fail the sass linter
- `git add .`
- `git commit`
- Check that the commit failed
### Reviews

@Automattic/sdev-feed
